spec/requests/rack_attack_spec.rb

```diff
@@ -0,0 +1,242 @@
+      end
+
+      it 'success status' do
+        expect(response.status).to eq(200)
+      end
+
+      it 'does not track the ip' do
+        expect(request.env['rack.attack.matched']).to eq('allow from localhost')
+        expect(request.env['rack.attack.match_type']).to eq(:safelist)
+        expect(request.env['rack.attack.match_data']).to_not be_present
+        expect(request.env['REMOTE_ADDR']).to eq('127.0.0.1')
+      end
+    end
+
+    describe 'localhost ::1' do
+      before do
+        get '/', {}, 'REMOTE_ADDR' => '::1'
+      end
+
+      it 'success status' do
+        expect(response.status).to eq(200)
+      end
+
+      it 'does not track the ip' do
+        expect(request.env['rack.attack.matched']).to eq('allow from localhost')
+        expect(request.env['rack.attack.match_type']).to eq(:safelist)
+        expect(request.env['rack.attack.match_data']).to_not be_present
+        expect(request.env['REMOTE_ADDR']).to eq('::1')
+      end
+    end
+
+    describe 'NU network 165.214.0.0/16' do
+      before do
+        get '/', {}, 'REMOTE_ADDR' => '165.124.124.32'
+      end
+
+      it 'success status' do
+        expect(response.status).to eq(200)
+      end
+
+      it 'does not track the ip' do
+        expect(request.env['rack.attack.match_type']).to eq(:safelist)
+        expect(request.env['rack.attack.match_data']).to_not be_present
+        expect(request.env['REMOTE_ADDR']).to eq('165.124.124.32')
+      end
+    end
+
+    describe 'NU network 129.105.0.0/16' do
+      before do
+        get '/', {}, 'REMOTE_ADDR' => '129.105.215.146'
+      end
+
+      it 'success status' do
+        expect(response.status).to eq(200)
+      end
+
+      it 'does not track the ip' do
+        expect(request.env['rack.attack.match_type']).to eq(:safelist)
+        expect(request.env['rack.attack.match_data']).to_not be_present
+        expect(request.env['REMOTE_ADDR']).to eq('129.105.215.146')
+      end
+    end
+  end
+
+  context 'throttle subnets' do
+    describe 'not signed in user' do
+      context 'all paths' do
+        before do
+          @epoch_time = Time.now.to_i
+          get '/', {}, 'REMOTE_ADDR' => '1.2.3.4'
+        end
+
+        it 'counts the subnet' do
+          key = "rack::attack:#{(Time.now.to_i/@period).to_i}:req/subnet:1.2.3."
+          expect(Rack::Attack.cache.store.read(key)).to eq(1)
+        end
+
+        it 'tracks the subnet' do
+          data = { :count => 1, :limit => 100, :period => @period.to_i, :epoch_time => @epoch_time }
+          expect(request.env['rack.attack.throttle_data']['req/subnet']).to eq(data)
+          expect(request.env['REMOTE_ADDR']).to eq('1.2.3.4')
+        end
+
+        it 'suppresses the subnet' do
+          key = "rack::attack:#{(Time.now.to_i/@period).to_i}:req/subnet:1.2.3."
+          Rack::Attack.cache.store.increment(key, 99)
+          @epoch_time = Time.now.to_i
+          get '/about', {}, 'REMOTE_ADDR' => '1.2.3.5'
```

Comment:
minor: add new line above `get` and remove new line between `data` and `expect`

Reply:
Done.